receiver/k8sobjectsreceiver/config.go

```diff
@@ -65,7 +65,7 @@ func (c *Config) Validate() error {
 	for _, object := range c.Objects {
 		gvrs, ok := validObjects[object.Name]
 		if !ok {
-			availableResource := make([]string, len(validObjects))
+			availableResource := make([]string, 0, len(validObjects))
```

Comment:
This use case would be better improved if we set via an index instead of append. 

Leaving `availableResource := make([]string, len(validObjects))` but doing `availableResource[i] = k` feels the most efficient, since we know that every element in `validObjects` needs to end up in `availableResource`.

Reply:
we know the validObjects is a map, so if we wanna set the value to the index of availableResource, we need a counter there, IMO append is good

Reply:
Do we actually need to update availableResource @TylerHelmuth ?